Updated the extensions spec with the new discover routes

mnltejaswini opened the pull request:
Updated the extensions spec as per the comments on the PR . The major changes are 
- Changed the discovery API route to /v2/_ext/discover (https://github.com/opencontainers/distribution-spec/pull/111/files/a9c3cdc9e554894edc30dfea5e808406244a93be#r728376000 )
- Changed the extension response to contain an array of objects rather than strings (https://github.com/opencontainers/distribution-spec/pull/111/files/a9c3cdc9e554894edc30dfea5e808406244a93be#r728380290)

Signed-off-by: Tejaswini Duggaraju <naduggar@microsoft.com>

mnltejaswini committed:
Updated the extensions spec with the new discover routes

Signed-off-by: Tejaswini Duggaraju <naduggar@microsoft.com>

sajayantony commented:
Did we discuss about reserving `_oci` ?

mnltejaswini commented:
Yes, both _ext and _oci are considered as reserved namespaces. I am checking for them in the distribution's extension PR

mnltejaswini committed:
Added a section for reserved namespaces

Signed-off-by: Tejaswini Duggaraju <naduggar@microsoft.com>

stevvooe commented on ext/README.md:
```diff
@@ -47,10 +47,13 @@ can be access under a repository.
 
-Extention definitions SHOULD be placed under `./ext/`. Extension files
-SHOULD follow the `ext-$name.md`. Refer [ext.md](./ext.md) for more details.
+Extension definitions SHOULD be placed under `./ext/`. Extension files
```
Might clarify that this applies to the layout of the specification.

```suggestion
Extension definitions SHOULD be placed under `./ext/` in the specification repository. Extension files
```

stevvooe commented on ext/ext.md:
```diff
@@ -31,14 +31,16 @@ Content-Type: application/json
     "extensions: [
-        "_<ns>/<ext>/<component>",
+        {
+            "name": "_<ns>/<ext>/<component>"
```
Above, we have `_ext/discover`, which would imply this should be `_<ext>/<operation>`. Was there a reason to have the extra path component?

mnltejaswini replied:
As per my understanding, extensions are namespaces and components are contained under an extension. Hence the path ```_ns/extension/component```. If the discover API as to be aligned with this model, one proposal could be ```_oci/ext/discover``` @stevvooe what is your guidance here?

stevvooe replied:
Per the discussion in the OCI meeting, we are going with `/v2/_<ext>/<operation>` and `/v2/<name>/_<ext>/<operation>`.

vbatts commented on ext/ext.md:
```diff
@@ -31,14 +31,16 @@ Content-Type: application/json
 {
     "extensions: [
-        "_<ns>/<ext>/<component>",
+        {
```
Curious why to have a hashmap inside an array here?

mnltejaswini replied:
We updated the response to be an array of extension objects rather than just the strings. This will enable us to add more details about an extension in the response if needed in the future. /cc @sudo-bmitch

vbatts replied:
ah ok. might be good for me to include a basic golang source with the structs for easy marshalling of this.

vbatts commented on ext/ext.md:
````diff
@@ -12,13 +12,13 @@ extensions.
 
 ```HTTP
-    GET /v2/_ext/
+    GET /v2/_ext/discover
````
I looked over the linked discussion, but I'm still not understanding why to make the discovery into an endpoint rather than just at the base `/v2/_ext/`?

vbatts commented on ext/ext.md:
```diff
@@ -2,7 +2,7 @@
 ## Summary
 
-This base extension is used to discover and return an array of extensions.
+This base extension namespace is used to discover and return an array of extensions.
```
This namespace concept needs some words of definition what it is namespacing, right?

mnltejaswini replied:
@vbatts  My understanding is it is namespacing the extensions supported by distribution. The base extension namespace includes discover component that lists all the available extensions.

sajayantony replied:
@vbatts @mnltejaswini can we get the baseline in and reword since we agreed to get this in some form and iterate with updates. 
I think we need this to make progress on the implementation PR and ORAS path updates as well.

mnltejaswini committed:
Update ext/README.md

Addressed the comment for indicating that the extension definitions applies to the layout of the spec

Co-authored-by: Stephen Day <stevvooe@users.noreply.github.com>

vbatts commented:
TODO(vb) define the object in the `extensions` list
